### PR TITLE
feat(oas) - accurate model OAS representation - R to U

### PR DESCRIPTION
**Pull request by patrick-medusajs**
### Scope

Models R to U

### What

Refactor OAS for models to accurately represent their shape in API responses.

### Why

About 33% of model fields are not accurately represented in the OAS. Most of the issues are:
- fields that can not be omitted in the response are not declared as `required`
- fields that could return `null` as their value are not declared as `nullable: true`

When using a code generator, these OAS issues would lead to inaccurate response shapes in the generated client.

#### nullable
Fields meeting at least one of the following condition will be represented as `nullable: true` in OAS:
* The field is decorated with `@Column({ nullable: true })`
* The field is decorated with `@OneToOne`, `@ManyToOne`
* The field is decorated with `@DeleteDateColumn`

#### optional
Fields meeting at least one of the following conditions will never be listed as `required` in OAS and will be considered optional and could be omitted in the response:
* The field is decorated with `@OneToOne`, `@ManyToOne`, `@OneToMany`, `@ManyToMany`
* The field is decorated with `@FeatureFlagColumn`
* The field is decorated with `@Column({select: false})`
* The field is representing dynamic values not persisted in the database

Fields not meeting any of the conditions above will be declared as `required` and are expected to be present in the response.

### Test
* Ran OAS validator.
* Ran docs build script.

Expect OAS changes to be reflected in the API documentation.

### CODE REVIEW

**Review by patrick-medusajs — COMMENTED** [review]
self-review: self-review: I'd recommend using the [Model TS refactor PR](https://github.com/medusajs/medusa/pull/3202/files) to facilitate the review process. I've also create a [demo PR](https://github.com/medusajs/medusa/pull/3204/files) to showcase the generate types from the OAS.

Note that the OAS demo PR does not contain the latest changes to `sales-channel.ts` and `sales-channel-location.ts`

**Review by shahednasser — CHANGES_REQUESTED** [review]

File: packages/medusa/src/models/region.ts [hunk 1]
```diff
@@ -161,11 +172,13 @@ export class Region extends SoftDeletableEntity {
...
- *     type: string
  *     description: The ID of the tax provider used in this region
+ *     nullable: true
+ *     type: string
  *     example: null
  *   tax_provider:
  *     description: Available if the relation `tax_provider` is expanded.
+ *     nullable: true
  *     $ref: "#/components/schemas/TaxProvider"
  *   payment_providers:
```
[the remark this answers is not in the document]
Comment (patrick-medusajs) [reply packages/medusa/src/models/region.ts:183]: Fixed in dddaaba4c7148527ef52d29477c0507541937192

File: packages/medusa/src/models/return-reason.ts [hunk 2]
```diff
@@ -91,11 +101,13 @@ export class ReturnReason extends SoftDeletableEntity {
  *     description: "The date with timezone at which the resource was updated."
```
[the remark this answers is not in the document]
Comment (patrick-medusajs) [reply packages/medusa/src/models/return-reason.ts]: Fixed in dddaaba4c7148527ef52d29477c0507541937192

File: packages/medusa/src/models/sales-channel-location.ts [hunk 3]
```diff
@@ -30,27 +30,39 @@ export class SalesChannelLocation extends SoftDeletableEntity {
  * title: "Sales Channel Stock Location"
  * description: "Sales Channel Stock Location link sales channels with stock locations."
  * type: object
+ * required:
+ *   - created_at
+ *   - id
+ *   - location_id
+ *   - sales_channel_id
+ *   - updated_at
```
[the remark this answers is not in the document]
Comment (patrick-medusajs) [reply packages/medusa/src/models/sales-channel-location.ts]: Fixed in dddaaba4c7148527ef52d29477c0507541937192

File: packages/medusa/src/models/shipping-method-tax-line.ts [hunk 4]
```diff
@@ -35,43 +35,51 @@ export class ShippingMethodTaxLine extends TaxLine {
...
  *     description: The line item tax line's ID
- *     example: smtl_01G1G5V2DRX1SK6NQQ8VVX4HQ8
- *   shipping_method_id:
  *     type: string
- *     description: The ID of the line item
- *     example: sm_01F0YET7DR2E7CYVSDHM593QG2
- *   shipping_method:
- *     description: Available if the relation `shipping_method` is expanded.
- *     $ref: "#/components/schemas/ShippingMethod"
+ *     example: litl_01G1G5V2DRX1SK6NQQ8VVX4HQ8
```
[the remark this answers is not in the document]
Comment (patrick-medusajs) [reply packages/medusa/src/models/shipping-method-tax-line.ts]: Fixed in dddaaba4c7148527ef52d29477c0507541937192

**Comment by patrick-medusajs** [comment]
@shahednasser thank you for reviewing these PR. Much appreciated 🙌

**Pull request merged**